src/Compilers/CSharp/Test/Semantic/Semantics/InlineArrayTests.cs

```diff
@@ -0,0 +1,6195 @@
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+struct Buffer
+{
+}
+";
+            var comp = CreateCompilation(src + InlineArrayAttributeDefinition);
+            CompileAndVerify(comp, symbolValidator: verify, sourceSymbolValidator: verify).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                var buffer = m.GlobalNamespace.GetTypeMember("Buffer");
+
+                Assert.True(buffer.HasInlineArrayAttribute(out int length));
+                Assert.Equal(10, length);
+                Assert.False(buffer.TryGetInlineArrayElementType().HasType);
+            }
+        }
+
+        [Fact]
+        public void InlineArrayType_10_WithStaticFields()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+struct Buffer
+{
+    public static long A;
+    private int _element0;
+    public static short B;
+}
+";
+            var comp = CreateCompilation(src + InlineArrayAttributeDefinition);
+            CompileAndVerify(comp, symbolValidator: verify, sourceSymbolValidator: verify).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                var buffer = m.GlobalNamespace.GetTypeMember("Buffer");
+
+                Assert.True(buffer.HasInlineArrayAttribute(out int length));
+                Assert.Equal(10, length);
+                Assert.Equal(SpecialType.System_Int32, buffer.TryGetInlineArrayElementType().SpecialType);
+            }
+        }
+
+        [Fact]
+        public void InlineArrayType_11_WithSingleStaticField()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+struct Buffer
+{
+    private static int _element0;
+}
+";
+            var comp = CreateCompilation(src + InlineArrayAttributeDefinition);
+            CompileAndVerify(comp, symbolValidator: verify, sourceSymbolValidator: verify).VerifyDiagnostics();
+
+            void verify(ModuleSymbol m)
+            {
+                var buffer = m.GlobalNamespace.GetTypeMember("Buffer");
+
+                Assert.True(buffer.HasInlineArrayAttribute(out int length));
+                Assert.Equal(10, length);
+                Assert.False(buffer.TryGetInlineArrayElementType().HasType);
+            }
+        }
+
+        [Fact]
+        public void InlineArrayType_12_Class()
+        {
+            var src = @"
+[System.Runtime.CompilerServices.InlineArray(10)]
+class Buffer
+{
+    private int _element0;
+}
+
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All, AllowMultiple = false)]
+    public sealed class InlineArrayAttribute : Attribute
+    {
+        public InlineArrayAttribute (int length)
+        {
+            Length = length;
+        }
+
+        public int Length { get; }
+    }
+}
+";
+            var comp = CreateCompilation(src, options: TestOptions.DebugDll.WithMetadataImportOptions(MetadataImportOptions.All));
```

Comment:
> `.WithMetadataImportOptions(MetadataImportOptions.All)`

Just curious - why is this needed in this test?

Reply:
> Just curious - why is this needed in this test?

By default, private fields in classes could be dropped on import. I do not want this to affect the test scenario.